static/app/views/dashboardsV2/detail.tsx

```diff
@@ -188,21 +187,29 @@ class DashboardDetail extends Component<Props, State> {
   };
 
   onRouteLeave = () => {
+    const {dashboard} = this.props;
+    const {modifiedDashboard} = this.state;
+
     if (
       ![DashboardState.VIEW, DashboardState.PENDING_DELETE].includes(
         this.state.dashboardState
-      )
+      ) &&
+      !isEqual(modifiedDashboard, dashboard)
```

Comment:
should changing the dimensions/position of the widgets trigger the 'unsaved changes' pop up when canceled?

Reply:
Oof, we don't actually show this alert in general when cancelling during the edit stage. Updated so that's fixed in this PR too